sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -605,15 +605,13 @@ object RapidsConf {
     .booleanConf
     .createWithDefault(false)
 
-  val ENABLE_CAST_STRING_TO_DECIMAL = conf("spark.rapids.sql.castStringToDecimal.enabled")
-    .doc("When set to true, enables casting from strings to decimal type on the GPU. Currently " +
-      "string to decimal type on the GPU might produce results which slightly differed from the " +
-      "correct results when the string represents any number exceeding the max precision that " +
-      "CAST_STRING_TO_FLOAT can keep. For instance, the GPU returns 99999999999999987 given " +
-      "input string \"99999999999999999\". The cause of divergence is that we can not cast " +
-      "strings containing scientific notation to decimal directly. So, we have to cast strings " +
-      "to floats firstly. Then, cast floats to decimals. The first step may lead to precision " +
-      "loss.")
+  val ENABLE_LIMITED_NEGATIVE_DECIMAL_SCALE =
```

Comment:
If this is a bug in Spark, then I would rather that we just document that we do not have the same bug, instead of adding a new config.  Also this config is really confusing because it does not mention cast anywhere in it. So users have no idea how we are partially compatible. If we feel we should keep the config, then it needs to be rewritten to express exactly how we are different.

Reply:
We can rename the config to whatever makes the most sense. Do we need the config though? I think we do because if I, as a user, am accelerating a query that is casting a String to Decimal and has a negative scale, the results will not match the CPU spark. How else do we make sure the user knows about this? Is documentation enough? If we decide it is enough then I will remove the config.

Reply:
We are trying to be bug for bug compatible with Spark. But it is for negative scale decimals. They are off by default and cannot be written to/read from ORC or Parquet. Also you are the one that found the bug because no one has upgraded to that version yet, or not one is using negative scale decimals. All of that together makes me okay with just documenting that we do it differently.  I would also be fine with just falling back to the CPU 100% of the time for negative scale decimals on these versions of Spark and documenting why. To me the config is just too much overhead to worry about. But you are the one writing this patch, so I will defer to you on that. But if you do have a config it has to be very clear about it being only cor casting negative scale decimal values to strings. In the name of the config. In the documentation for the config. and in the naming of the code/shim layer for all of that too.